### PR TITLE
MEN-2805: Document the bare minimum upgrade from OS to ENT.

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

**Comment by kacf** [comment]
@estenberg: I fear that I may be thinking too much like a programmer when writing, and creating links between the documents. But it's hard to say how it reads with fresh eyes. Can you tell me if seems ok? The alternative is to duplicate the guide sections, but there is quite a lot of text (explaining what tenant tokens are, for example).

File: 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+<!-- AUTOMATION: execute=docker volume create --name=mender-redis-db -->
+<!-- AUTOMATION: execute=docker volume inspect --format '{{.Mountpoint}}' mender-artifacts -->
+<!-- AUTOMATION: execute=sed -i.bak 's/MINIO_ACCESS_KEY:.*/MINIO_ACCESS_KEY: Q3AM3UQ867SPQQA43P2F/' config/prod.yml -->
+<!-- AUTOMATION: execute=sed -i.bak 's/MINIO_SECRET_KEY:.*/MINIO_SECRET_KEY: abcssadasdssado798dsfjhkksd/' config/prod.yml -->
+<!-- AUTOMATION: execute=sed -i.bak 's/DEPLOYMENTS_AWS_AUTH_KEY:.*/DEPLOYMENTS_AWS_AUTH_KEY: Q3AM3UQ867SPQQA43P2F/' config/prod.yml -->
+<!-- AUTOMATION: execute=sed -i.bak 's/DEPLOYMENTS_AWS_AUTH_SECRET:.*/DEPLOYMENTS_AWS_AUTH_SECRET: abcssadasdssado798dsfjhkksd/' config/prod.yml -->
+<!-- AUTOMATION: execute=./run up -d -->
+
+This section describes how to upgrade from the Mender Open Source server, to the
+Mender Enterprise server.
```
Comment (estenberg) [inline 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md]: Would add something like "If you have any questions or would like assistance with upgrading to Mender Enterprise, please email [contact@mender.io](mailto:contact@mender.io)."

File: 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+docker volume create --name=mender-artifacts-backup
+docker volume create --name=mender-db-backup
+docker volume create --name=mender-elasticsearch-db-backup
+docker volume create --name=mender-redis-db-backup
+```
+
+Clone the contents of the original volumes to the backup volumes:
+
+```bash
+docker run --rm -ti -v mender-artifacts:/from        -v mender-artifacts-backup:/to        alpine cp -a /from /to
```
Comment (estenberg) [inline 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md]: what is /from and /to?

If they are directories that might vary depending on how the server was set up I would recommend to set and use shell variables (like we do for IP addresses in other parts of the doc).

File: 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+   fetched in the previous step
+
+   <!-- AUTOMATION: execute=sed -i -e "s;DEVICEAUTH_DEFAULT_TENANT_TOKEN:;DEVICEAUTH_DEFAULT_TENANT_TOKEN: $TENANT_TOKEN;" config/enterprise.yml -->
+
+4. Start the server again:
+   ```bash
+   ./run up -d
+   ```
+
+5. Continue with [the Enterprise
```
Comment (estenberg) [inline 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md]: I think this is fine, as it is the last step of this tutorial (no back and forth). However, the word "Continue" sounds like I should have started with the other tutorial.

Also, just wanted to make sure the URL is right, will "..#saving-the-enterprise-configuration" redirect to the section of the other tutorial?

Perhaps "To finalize the upgrade follow the steps in the [Enterprise installation tutorial from saving the enterprise configuration](..#saving-the-enterprise-configuration)."

File: 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+## Migrating clients
+
+With Mender Enteprise, each device needs to have a tenant token associated with
+it. When upgrading from Open Source however, the devices that are already
+deployed will be lacking this token. Therefore we need to use a special default
+tenant token which will automatically be assigned to all clients that try to
+connect without a tenant token.
+
+1. If you haven't already, [fetch the **tenant
```
Comment (estenberg) [inline 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md]: If I am starting on this doc, then I have not already done this, right? This makes me a bit insecure if I should first have followed the installation tutorial...

I would actually recommend to duplicate this command here.

File: 07.Administration/06.Upgrading/docs.md [hunk 5]
```diff
@@ -8,6 +8,10 @@ This is a guide for upgrading the Mender Server in production environments. In
 general updating is only supported between connected minor versions, and latest
 minor to next major version, and this guide reflects this.
```
[the remark this answers is not in the document]
Comment (kacf) [reply 07.Administration/06.Upgrading/docs.md]: I added a section now, so we don't forget. There should not be any difference in the steps.

File: 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md (hunk 3 above)
Comment (kacf) [reply 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md]: Updated! And yes, I've verified that the link works!

File: 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md (hunk 4 above)
Comment (kacf) [reply 07.Administration/02.Production-installation/01.Upgrading-from-OS-to-Enterprise/docs.md]: Ok. You need the tenant ID from the `create-user` command though, and that command is not repeatable. But after that command, the guide specifically tells you to save the tenant ID, so hopefully people will do that.

**Comment by kacf** [comment]
Thanks for the review! All issues resolved!

**Pull request merged**